Expand role of CaptureRequest and add Accept tool

veracioux opened the pull request:
- `CaptureRequest` class and `CaptureRequest::ExportTask` have been repurposed so they can be used for all actions that can potentially close the GUI in the future.
- Added accept button that is used to accept all capture requests that were specified from the CLI. After that the GUI closes. This will provide an elegant way to specify any action as CLI arguments (e.g. --pin, --upload, etc)
- Save, copy and imgur-upload buttons are disabled while the accept button is active
- Redesigned help text (OverlayMessage)
  - More readable format
  - Tracks configured shortcuts
- Removed `graphicCapture`, `fullScreen` and `captureScreen` DBus methods and replaced them all with a single `requestCapture` method. This method takes a byte array representation of a `CaptureRequest`. This way we don't have to recreate `CaptureRequest`s in `FlameshotDBusAdapter` which was a duplication of effort.

Closes #1336.
Closes #1416.
Closes #1531.
Closes #1813.
Closes #1897.

veracioux commented:
Closes #1336.

mmahmoudian commented:
While testing the "accept" functionality, I faced a problem which does not happen with the b650231f commit.

```sh
./Flameshot-0.10.1.x86_64.AppImage gui --raw | display
```

> ```
> display: no decode delegate for this image format `' @ error/constitute.c/ReadImage/572.
> ```

I don't know what the issue is, but it seems something is the image format is not the same as the older commit.

veracioux commented:
@mmahmoudian
I didn't mind the current appearance of the message box so I didn't bother to change anything. On my system, I do get a border around the message - maybe this has to do with the system theme. And I do get some transparency. I'll try to make it consistent. Rounded corners are fine and will take but a few seconds to implement.

I'll investigate the accept issue.

mmahmoudian commented:
I compiled against 1d334d372bbe9423f0a2b2252a9930f4dd5afc74 and I can confirm that the issue with `--raw` disappeared. I haven't looked into the AppImage and I'm significantly under-qualified to comment on what is happening there.

But found few issues:
1. the save button and copy to clipboard button are disappeared despite of being marked to be shown:
    ![image](https://user-images.githubusercontent.com/390889/136784029-982ed76e-7e84-48d4-ac16-c9ea29b73a4f.png)
2. the <kbd>Enter</kbd> terminates the capture when `flameshot gui` is running. In my head (I might be wrong) it should work so that in case of `flameshot gui` it shows the "final action buttons" (save to file, upload, copy to clipboard, ...), but when some "final action flags" are provided (e.g `--raw`, `--clipboard`, `--path`), then all those "final action buttons" should be disabled and also disappear and instead the "accept capture button" be shown.

What do you think?

veracioux commented:
@mmahmoudian

- Fixed the button issue when launched from systray
- The message now disappears only after you make a selection
- Changed the overlay message style as per your suggestions

Is that it?

veracioux commented:
@mmahmoudian Done. Ready for review?
